internal/pagefile_test.go

```diff
@@ -3,75 +3,262 @@ package internal_test
 import (
 	"bytes"
 	"fmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/utility"
 	"io"
 	"io/ioutil"
 	"log"
 	"os"
 	"testing"
-
-	"github.com/stretchr/testify/assert"
-	"github.com/wal-g/wal-g/internal"
-	"github.com/wal-g/wal-g/utility"
 )
 
 const (
-	pagedFileName        = "../test/testdata/base_paged_file.bin"
-	sampeLSN      uint64 = 0xc6bd4600
+	pagedFileName               = "../test/testdata/base_paged_file.bin"
+	pagedFileSizeInBytes        = 65536
+	pagedFileBlockCount         = pagedFileSizeInBytes / internal.DatabasePageSize
+	sampleLSN            uint64 = 0xc6bd4600
+	smallLSN             uint64 = 0
+	bigLSN                      = sampleLSN * 2
 )
 
-// In this test we use actual postgres paged file which
+var regularTestIncrement = newTestIncrement(sampleLSN)
+var smallLsnTestIncrement = newTestIncrement(smallLSN)
+var bigLsnTestIncrement = newTestIncrement(bigLSN)
+
+// In this test series we use actual postgres paged file which
 // We compute increment with LSN taken from the middle of a file
 // Resulting increment is than applied to copy of the same file partially wiped
 // Then incremented file is binary compared to the origin
 func TestIncrementingFile(t *testing.T) {
-	loclLSN := sampeLSN
-	postgresFileTest(loclLSN, t)
-
+	postgresApplyIncrementTest(regularTestIncrement, t)
 }
 
 // This test covers the case of empty increment
 func TestIncrementingFileBigLSN(t *testing.T) {
-	loclLSN := sampeLSN * 2
-	postgresFileTest(loclLSN, t)
-
+	postgresApplyIncrementTest(bigLsnTestIncrement, t)
 }
 
-// This test converts the case when increment is bigger than original file
+// This test covers the case when increment is bigger than original file
 func TestIncrementingFileSmallLSN(t *testing.T) {
-	loclLSN := uint64(0)
-	postgresFileTest(loclLSN, t)
+	postgresApplyIncrementTest(smallLsnTestIncrement, t)
+}
+
+func postgresApplyIncrementTest(testIncrement *TestIncrement, t *testing.T) {
+	incrementReader := testIncrement.NewReader()
+	tmpFileName := pagedFileName + "_tmp"
+	copyFile(pagedFileName, tmpFileName)
+	defer os.Remove(tmpFileName)
+	tmpFile, _ := os.OpenFile(tmpFileName, os.O_RDWR, 0666)
+	tmpFile.WriteAt(make([]byte, 12345), 477421568-12345)
+	tmpFile.Close()
+	err := internal.ApplyFileIncrement(tmpFileName, incrementReader, false)
```

Comment:
I can't say I enjoy this existing test too, but since it is going to be deleted anyway after integration of reverse delta unpack, maybe we should leave it as it is